docs/development.md

```diff
@@ -180,3 +180,30 @@ make test-e2e PARALLEL=4
 
 This will run only 4 tests in parallel. By default, it is set to the value of
 `GOMAXPROCS`.
+
+### spec.go conventions
+
+- Add explantion on top of each struct and struct field in `spec.go` to explain what it does,
+so that when OpenAPI spec is auto-generated it shows up there.
+
+- Structs that are referred in any other struct in the form of array should have a comment
+of the format `// kedgeSpec: io.kedge.*`, where `*` is name of that struct. This becomes the
+identity or refernce of that struct in OpenAPI specification.
+
+- If in any struct you are embedding other struct there is no need to add explanation
```

Comment:
rephrase

maybe: `If you are embedding a struct, there is no need to add an explanatory comment. `